### PR TITLE
Ct 1852 Accept/Reject overturned SAR cases

### PR DESCRIPTION
**Pull request by aliuk2012**

### CODE REVIEW

File: spec/site_prism/page_objects/sections/cases/ico/ico_decision_section.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+module PageObjects
+  module Sections
+    module Cases
+      module ICO
+        class IcoDecisionSection < SitePrism::Section
```
Comment (misaka) [inline spec/site_prism/page_objects/sections/cases/ico/ico_decision_section.rb]: Can we keep this consistent and use `ICODecisionSection`?

File: app/views/shared/components/_case_page_original_case_details.html.slim [hunk 2]
```diff
@@ -1,4 +1,4 @@
-- if @case.ico?
+- if @case.ico? || @case.overturned_ico?
```
Comment (misaka) [inline app/views/shared/components/_case_page_original_case_details.html.slim]: We're using different styles of detecting whether to render bits in views. In `edit.html.slim` we use:

```
  - if @case.respond_to?(:ico_decision) && @case.ico_decision.present?
    ...
```

Which is more in-line with the duck-typing philosophy in Ruby. Feels like we should choose one style and stick with it, what do you think? If we wanted we could use duck-typing here with:

```
  - if @case.respond_to?(:original_case) && @case.original_case.present?
    ...
```

File: spec/site_prism/page_objects/sections/cases/ico/ico_decision_section.rb (hunk 1 above)
Comment (aliuk2012) [reply spec/site_prism/page_objects/sections/cases/ico/ico_decision_section.rb]: Good point

File: app/views/shared/components/_case_page_original_case_details.html.slim (hunk 2 above)
Comment (aliuk2012) [reply app/views/shared/components/_case_page_original_case_details.html.slim]: I agree with you, I completely forgot we could inspect original_case for both case types.

**Review by beccapearce — APPROVED** [review]

**Review by misaka — APPROVED** [review]

**Pull request merged**